backend/src/main/scala/bloop/logging/ProcessLogger.scala

```diff
@@ -0,0 +1,87 @@
+package bloop.logging
+
+import java.io.{
+  BufferedReader,
+  ByteArrayOutputStream,
+  InputStream,
+  InputStreamReader,
+  OutputStream,
+  PrintStream
+}
+import scala.annotation.tailrec
+
+/**
+ * Logs the output on stdout and std err of `process` to the `underlying` logger
+ *
+ * @param underlying The logger that receives the output of the process
+ * @param process    The process whose output must be logged.
+ */
+class ProcessLogger(underlying: Logger, process: Process) {
+  private[this] val processOut = process.getInputStream
+  private[this] val processErr = process.getErrorStream
+
+  def start(): Unit = {
+    underlying.debug("Starting to log output from process...")
+    new StreamLogger(underlying.info, processOut).start()
+    new StreamLogger(underlying.error, processErr).start()
+  }
+}
+
+object ProcessLogger {
+
+  private val encoding = sys.props("file.encoding")
+
+  /**
+   * Creates an `OutputStream` from a logging function
+   *
+   * @param logFn A function that logs the given string to a logger.
+   * @return An `OutputStream` that can be used as `System.in` or `System.err`.
+   */
+  def toOutputStream(logFn: String => Unit): OutputStream = {
+    val outputStream = new OutputStream {
+      private val buffer = new ByteArrayOutputStream
+
+      override def write(x: Int): Unit = synchronized {
+        if (x == '\n') flush()
+        else buffer.write(x)
+      }
+
+      override def flush(): Unit = synchronized {
+        val bytes = buffer.toByteArray()
+        buffer.reset()
+        val content = new String(bytes, encoding)
+        logFn(content)
+      }
+    }
+    outputStream
+  }
+
+  /**
+   * Creates a `PrintStream` from a logging function
+   *
+   * @param logFn A function that logs the given string to a logger.
+   * @return A `PrintStream` that can be used as `System.in` or `System.err`.
+   */
+  def toPrintStream(logFn: String => Unit): PrintStream = {
+    val outputStream = toOutputStream(logFn)
+    new PrintStream(outputStream, /* autoflush = */ false, encoding)
```

Comment:
Why have `autoFlush` as a comment and not use it directly as a named parameter?

Reply:
This is a Java method, named parameters don't work with those AFAIK